Example/ViewController.swift

```diff
@@ -58,13 +58,10 @@ struct EstablishmentsResponse {
 }
 
 extension EstablishmentsResponse: Parsable {
-    static func parse(from data: Data?, status: Int, headers: [String: String]?, errorParser: ErrorParsing.Type?, userInfo: [String: Any]?) -> Result<EstablishmentsResponse> {
-        return parse(from: data, status: status, headers: headers, errorParser: errorParser)
-    }
 
-    static func parse(from data: Data?, status: Int, headers: [String: String]?, errorParser: ErrorParsing.Type?) -> Result<EstablishmentsResponse> {
-        if status != 200 {
-            if let error = errorParser?.parseError(from: data, statusCode: status) {
+    static func parse(from data: Data?, errorParser: ErrorParsing.Type?, context: ParsableContext) -> Result<EstablishmentsResponse> {
```

Comment:
Context is probably more important than the error parser, so perhaps move it ahead. I wonder if there's a better name for the type that more succinctly describes what it is. Perhaps we should just have a `Response` struct that includes data, status code, headers, original request functionality?

Reply:
Very good point actually, `Response` makes more sense.